repos/system_upgrade/common/libraries/dnfconfig.py

```diff
@@ -0,0 +1,109 @@
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.common.config.version import get_source_major_version
+from leapp.libraries.stdlib import CalledProcessError, api
+
+
+def get_leapp_packages():
+    """
+    Return the list of leapp and leapp-repository rpms that should be preserved
+    during the upgrade.
+
+    It's list of packages that should be preserved, not what is really
+    installed.
+
+    The snactor RPM doesn't have to be installed, but if so, we have to take
+    care about that too as well to preven broken dnf transaction.
+    """
+    # TODO: should we set the seatbelt and exclude leapp RPMs from the target
+    # system too?
+    generic = ['leapp', 'snactor']
+    if get_source_major_version() == '7':
+        return generic + ['python2-leapp', 'leapp-upgrade-el7toel8']
+
+    return generic + ['python3-leapp', 'leapp-upgrade-el8toel9']
+
+
+def _strip_split(data, sep, maxsplit=-1):
+    """
+    Just like str.split(), but remove ambient whitespaces from all items
+    """
+    return [item.strip() for item in data.split(sep, maxsplit)]
+
+
+def _get_main_dump(context):
+    """
+    Return the dnf configuration dump of main options for the given context.
+
+    Returns the list of lines after the line with "[main]" section
+    """
+
+    try:
+        data = context.call(['dnf', 'config-manager', '--dump'], split=True)['stdout']
+    except CalledProcessError as e:
+        api.current_logger().error('Cannot obtain the dnf configuration')
+        raise StopActorExecutionError(
+            message='Cannot obtain data about the DNF configuration',
+            details={'stdout': e.stdout, 'stderr': e.stderr}
+        )
+
+    try:
+        # return index of the first item in the main section
+        main_start = data.index('[main]') + 1
+    except ValueError:
+        raise StopActorExecutionError(
+            message='Invalid DNF configuration data (missing [main])',
+            details=data,
+        )
+
+    output_data = {}
+    for line in data[main_start:]:
+        if '=' not in line:
+            # TODO: could we skip this check completely?
+            continue
```

Comment:
I did a little research:
1. The `config-manager` plugin dumps only the contents of `[main]` section when no repos supplied on command line. The code path that handles `--dump` is here: [DNF plugin code](https://github.com/rpm-software-management/dnf-plugins-core/blob/master/plugins/config_manager.py#L147).
2. The plugins uses `self.base.conf.dump()` to display the contents of the section ([link to implementation](https://github.com/rpm-software-management/dnf/blob/master/dnf/conf/config.py#L161)). The implementation outputs lines with `Key=Value` format (the `=` is always present), therefore, we should not need to check it.

Reply:
I've changed the implementation a little bit. Despite the current DNF implementation, I am worried the print could be in future changed and we could miss that. So instead of if-continue, added try/except block with a warning and comment.